### DISCUSSION TITLE
DAI 2023-2024 - Practical work 2 - Lionel Pollien and Arthur Menétrey

### DISCUSSION THREAD

**Discussion by Xenogix** [heig-vd-dai-course/heig-vd-dai-course#144] [Show and tell]
During this work we will create a simple protocole to play the famous "four in a row" game.

This is the current concept art of the game :

```
P1  <-  P2

 1 2 3 4 5 6 7
| | | | | | | | A
| | | | | | | | B
| | | | | | | | C
| | | |o| | | | D
| | |o|x| | | | E
| | |x|x|o| | | F
‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
```

Commands we think should be implemented :

Server side :

- ACCEPT                       => Accept a player JOIN request
- REJECT <message>    => Reject a request with an error message
- STATE <state object> => Send the state of the game to each player
- ABORT <message>    => Inform that the server will quit for the given message

Client side :

- JOIN <GUID> <username> => Ask to join a server game with an unique identifier
- PLACE <GUID> <column>  => Ask the server to place a pawn in the given column
- FF <GUID>                           => Forfeit the game
- QUIT <GUID>                      => Inform the server that the client will quit

We don't really for now what's the best way to deal with player identification. We wrote GUID as command parameters to deal with it but this may change.

Edge / Error cases :

Play may crash
Server may crash
Too much player

**Comment by Xenogix** [comment]
You can find our repo at this address : https://github.com/poLionel/DAI-PracticalWork2-LP-AM

**Reply by ludelafo** [reply]
I guess the commit hash is <https://github.com/poLionel/DAI-PracticalWork2-LP-AM/commit/c9b5b9b63eec28535c71d0d0bcfea0d2aa55f417>? Please do not forget to submit as mentioned in the course material.

**Reply by poLionel** [reply]
We confirm the hash is c9b5b9b. Sorry for the inconvenience...
